packages/xo-web/src/xo-app/backup/health/index.js

```diff
@@ -83,9 +161,122 @@ const ACTIONS = [
   vms: createGetObjectsOfType('VM'),
 })
 export default class Health extends Component {
+  componentDidUpdate(prevProps) {
+    const { jobs, vms, remotes, schedules } = this.props
+    if (
+      !isEqual(
+        map(prevProps.remotes, 'id').sort(),
+        map(remotes, 'id').sort()
+      ) ||
+      !isEqual(map(prevProps.vms, 'id').sort(), map(vms, 'id').sort()) ||
+      !isEqual(map(prevProps.jobs, 'id').sort(), map(jobs, 'id').sort()) ||
+      !isEqual(
+        map(prevProps.schedules, 'id').sort(),
+        map(schedules, 'id').sort()
+      )
+    ) {
+      this._setDetachedBackups()
+    }
+  }
+
+  _deleteBackups = backups => {
+    const nBackups = backups.length
+    return confirm({
+      title: _('deleteBackups', { nBackups }),
+      body: _('deleteBackupsMessage', { nBackups }),
+      icon: 'delete',
+    })
+      .then(() => deleteBackups(backups), noop)
+      .then(this._setDetachedBackups)
+  }
+
+  _detachedBackupActions = [
+    {
+      handler: this._deleteBackups,
+      icon: 'delete',
+      label: backups => _('deleteBackups', { nBackups: backups.length }),
+      level: 'danger',
+    },
+  ]
+
+  _getDetachedBackups = async () => {
```

Comment:
`_getDetachedBackups` is only used one time, it can be merged with `_setDetachedBackups`.